### PR TITLE
RSDK-5149 - motion.MoveOnMap updates

### PR DESCRIPTION
**Pull request by nicksanford**
[Ticket](https://viam.atlassian.net/browse/RSDK-5149)

### Depends on:
1. API: https://github.com/viamrobotics/api/pull/439
2. Python SDK: https://github.com/viamrobotics/viam-python-sdk/pull/530
3. RDK/GO SDK: https://github.com/viamrobotics/rdk/pull/3479

### Changes:
1. Changes motion.MoveOnMap to new signature defined in dependent PRs
3. Updates motion.GetPlan & motion.ListPlanStatuses to indicate that their outputs both motion.MoveOnMap & motion.MoveOnGlobe  executions & plans.
4. Fixes incorrect units in plan_deviaion_m comment
5. Fixes links which were pointing to out of date definitions of Go SDK types.
6. Add `make prettierfix` to make it easier to remember how to get prettier to fix itself.

### CODE REVIEW

**Comment by nicksanford** [comment]
I'm not sure why htmltest is failing:
https://github.com/viamrobotics/docs/actions/runs/7642981804/job/20824881650?pr=2426:
```
+ cd /github/workspace
+ htmltest --version
htmltest v0.17.0
+ htmltest -c .htmltest.yml
htmltest started at 04:37:17 on dist
========================================================================
components/arm/index.html
  Non-OK status: 500 --- components/arm/index.html --> https://pkg.go.dev/go.viam.com/rdk/spatialmath#Pose
  request exceeded our ExternalTimeout --- components/arm/index.html --> https://pkg.go.dev/go.viam.com/rdk/components/arm#Arm
  Non-OK status: 500 --- components/arm/index.html --> https://pkg.go.dev/go.viam.com/rdk/spatialmath#Pose
  request exceeded our ExternalTimeout --- components/arm/index.html --> https://pkg.go.dev/go.viam.com/api/component/arm/v1#JointPositions
  request exceeded our ExternalTimeout --- components/arm/index.html --> https://pkg.go.dev/go.viam.com/api/component/arm/v1#JointPositions
appendix/contributing/index.html
  request exceeded our ExternalTimeout --- appendix/contributing/index.html --> https://exiftool.org/install.html
========================================================================
✘✘✘ failed in [6](https://github.com/viamrobotics/docs/actions/runs/7642981804/job/20824881650?pr=2426#step:7:7)m4[7](https://github.com/viamrobotics/docs/actions/runs/7642981804/job/20824881650?pr=2426#step:7:8).40[8](https://github.com/viamrobotics/docs/actions/runs/7642981804/job/20824881650?pr=2426#step:7:9)760[9](https://github.com/viamrobotics/docs/actions/runs/7642981804/job/20824881650?pr=2426#step:7:10)53s
6 errors in 377 documents
```

The pages which it is complaining about were not changed in my PR.

File: Makefile [hunk 1]
```diff
@@ -56,3 +56,6 @@ serve-dev-draft: setup
 
 serve-dev-future: setup
 	hugo server -F $(DEV_OPTIONS) $(SERVE_OPTIONS)
+
+prettierfix: setup
```
Comment (nicksanford) [inline Makefile:60]: I added this as I have not succeeded in configuring VSCode's prettier plugin & this seems like a much more reproducible way to get prettier to fix docs for folks that are only contributing to docs occasionally.

**Review by nfranczak — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (npentrel) [reply Makefile:60]: Thank you!

File: docs/mobility/motion/_index.md [hunk 2]
```diff
@@ -356,7 +356,27 @@ logger.Info("Orientation of myArm from the motion service:", myArmMotionPose.Pos
...
+
+Each successful `MoveOnMap()` call retuns a unique `ExecutionID` which you can use to identify all plans generated during the `MoveOnMap()` call.
+
+You can monitor the progress of the `MoveOnMap()` call by querying `GetPlan()` and `ListPlanStatuses()`.
+
+`MoveOnMap()` is intended for use with the [navigation service](/mobility/navigation/), providing autonomous indoor navigation for rover [bases](/components/base/).
+
+To use `MoveOnMap()`, your [SLAM service](/mobility/slam/) must implement `GetPointCloudMap()` and `GetPosition()`
+
+Make sure the [SLAM service](/mobility/slam/) you use supports usage of the following methods in its {{< glossary_tooltip term_id="model" text="model's" >}} implementation of the [SLAM service API](/mobility/slam/#api).
```
Comment (npentrel) [inline docs/mobility/motion/_index.md:374]:
```suggestion
Make sure the [SLAM service](/mobility/slam/) you use supports usage of the following methods in its {{< glossary_tooltip term_id="model" text="model's" >}} implementation of the [SLAM service API](/mobility/slam/#api):
```

File: docs/mobility/motion/_index.md [hunk 3]
```diff
@@ -647,7 +687,7 @@ Replans always create new plans.
 
 Replans share the `ExecutionID` of the previously executing plan.
 
-All repeated fields are in time ascending order.
+All repeated fields are in chronological order.
```
Comment (npentrel) [inline docs/mobility/motion/_index.md:690]: Thanks - good catch

**Review by npentrel — APPROVED** [review]
Thank you - this looks great!

**Review by raybjork — APPROVED** [review]

**Comment by npentrel** [comment]
@nicksanford just want to check in, when do you think this will merge - or rather just confirm this hasn't been forgotten about

**Comment by nicksanford** [comment]
> @nicksanford just want to check in, when do you think this will merge - or rather just confirm this hasn't been forgotten about

This will get merged when the next release candidate (v0.20.0) is promoted to stable. It has not been forgotten. I'll merge it when the RC gets promoted.

**Comment by nicksanford** [comment]
@npentrel now that v0.20.0 is out I'm going to merge this now

**Comment by npentrel** [comment]
Thank you!

**Pull request merged**